src/java.base/share/classes/java/util/jar/JarFile.java

```diff
@@ -837,26 +837,34 @@ private void initializeVerifier() {
     public synchronized InputStream getInputStream(ZipEntry ze)
         throws IOException
     {
-        maybeInstantiateVerifier();
-        if (jv == null) {
-            return super.getInputStream(ze);
-        }
-        if (!jvInitialized) {
-            initializeVerifier();
-            jvInitialized = true;
-            // could be set to null after a call to
-            // initializeVerifier if we have nothing to
-            // verify
-            if (jv == null)
+        Objects.requireNonNull(ze, "ze");
```

Comment:
Is the NPE specified?

Reply:
Yes it is specified in the  JarFile main paragraph

`Unless otherwise noted, passing a null argument to a constructor or method in this class will cause a NullPointerException to be thrown.`

ZipFile::getInputStream validates the argument as soon as it is passed to getInputStream.